sale_commission/models/settlement.py

```diff
@@ -33,6 +34,11 @@ def _default_currency(self):
     currency_id = fields.Many2one(
         comodel_name='res.currency', readonly=True,
         default=_default_currency)
+    company_id = fields.Many2one(
+        comodel_name='res.company',
+        default=lambda self: self.env.user.company_id,
+        required=True
+    )
```

Comment:
Shouldn't `company_id` be used in a record rule?

Reply:
@eLBati it could be added in order to add the 'automatic filtering' in multi-company environments but, as far as I know, it is not mandatory.
In particular, I added this field because it was necessary after the changes backported from v11 and there (https://github.com/OCA/commission/tree/11.0/sale_commission) I haven't found any record rule.

Reply:
I think we should add that record rules for proper multi-company support, which is indeed a pending topic.

Reply:
ok then I'll add them in a separate commit so we can easily forward port them

Reply:
OK, great. You should put a migration scripts for emptying `company_id` field for avoiding mis-behaviors when you have commissions in several companies, or even better, try an heuristic for deducing the company of each settlement.

Reply:
@pedrobaeza emptying `company_id` is not possible because it is `required`, I'm trying to find the correct company for each settlement

Reply:
@pedrobaeza @eLBati I added the record rules and the migration scripts as you requested, please have a look, thanks

Reply:
I'm seeing that the company is already present in 11.0: https://github.com/OCA/commission/blob/0661addc5f0d8e28840e68d637f6297b8a85ac41/sale_commission/models/settlement.py#L37